01.general/02.how-to-create-your-own-ad-filters/docs.en.md

```diff
@@ -2150,6 +2122,66 @@ This rule will be available for every platform except Safari extension, iOS, and
+
+<a id="debug-filtering-log"></a>
+### Filtering log
+
+Filtering log is an advanced tool that will be helpful mostly to filter developers. It lists all web requests that pass through AdGuard, gives you exhaustive information on each of them, offers multiple sorting options, and has other useful features.
+
+Depending on which AdGuard product you're using, Filtering log can be located in different places. 
+
+* In **AdGuard for Windows** you'll find it inside *Ad Blocker* tab or via the tray menu;
+* In **AdGuard for Mac** it's under *Settings > Advanced > Filtering log*;
+* In **AdGuard for Android** it's a separate item in the side menu;
+* In **AdGuard browser extensions** it's accessible from the *Miscellaneous* settings tab or by right-clicking the extension icon.
+
+> In **AdGuard for iOS** and in **AdGuard for Safari** Filtering log does not exist because of the way content blockers are implemented in Safari. AdGuard doesn't see the web requests and therefore can't display them.
+
+<a id="selectors-debugging-mode"></a>
+### Selectors debugging mode
+
+Sometimes, you might need to check the performance of a given selector or a stylesheet. In order to do it without interacting with javascript directly, you can use a special `debug` style property. When `ExtendedCss` meets this property, it enables the "debug"-mode either for a single selector or for all selectors depending on the `debug` value.
+
+**Debugging a single selector**
+```
+#$#.banner { display: none; debug: true; }
+```
+
+**Enabling global debug**
+```
+#$#.banner { display: none; debug: global; }
+```
+
+<a id="testing-extended-selectors"></a>
+#### Testing extended selectors
+
+To load ExtendedCss to a current page, copy and execute the following code in a browser console:
+```
+!function(E,x,t,C,s,s_){C=E.createElement(x),s=E.getElementsByTagName(x)[0],C.src=t,
+C.onload=function(){alert('ExtCss loaded successfully')},s.parentNode.insertBefore(C,s)}
+(document,'script','https://AdguardTeam.github.io/ExtendedCss/extended-css.min.js')
+```
+
+Alternatively, install an "ExtendedCssDebugger" userscript: https://github.com/AdguardTeam/Userscripts/blob/master/extendedCssDebugger/extended-css.debugger.user.js
+
+You can now use the `ExtendedCss` constructor in the global scope, and its method `ExtendedCss.query` as `document.querySelectorAll`.
+```
+var selectorText = "div.block[-ext-has='.header:matches-css-after(content: Anzeige)']";
+
+ExtendedCss.query(selectorText) // returns an array of Elements matching selectorText
+```
+
+<a id="debug-scriptlets"></a>
+### Debugging scriptlets
+
+If you're using AdGuard browser extension and want to debug a [scriptlet rule](https://kb.adguard.com/en/general/how-to-create-your-own-ad-filters#scriptlets), you can get additional information by simpy having the Filtering log opened. In that case, scriptlets will switch to debug mode and will write more information to the browser's console.
```

Comment:
Maybe we should add a more detailed description of debugging with scriptlets and add examples?

For example, these scriptlets can be useful for debugging:

```
debug-current-inline-script
debug-on-property-read
debug-on-property-write
log-addEventListener
log-eval
log
```